src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,150 @@
+                                                                                                       DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                                       isEnabledByDefault: false,
+                                                                                                       helpLinkUri: null,
+                                                                                                       customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolStartAction(ssac =>
+            {
+                var enumSymbol = (INamedTypeSymbol)ssac.Symbol;
+                if (enumSymbol.TypeKind != TypeKind.Enum)
+                {
+                    return;
+                }
+
+                var invalidMembersByValue = new ConcurrentDictionary<object, List<SyntaxNode>>();
+
+                // Collect duplicated values...
+                ssac.RegisterOperationAction(oc =>
+                {
+                    var fieldInitializerOperation = (IFieldInitializerOperation)oc.Operation;
+                    if (fieldInitializerOperation.InitializedFields.Length != 1 ||
+                        !fieldInitializerOperation.Value.ConstantValue.HasValue)
+                    {
+                        return;
+                    }
+
+                    var onlyReferencesOneField = GetFilteredDescendants(fieldInitializerOperation, op => op.Kind != OperationKind.Binary)
+                        .OfType<IFieldReferenceOperation>()
+                        .Any();
+                    if (onlyReferencesOneField)
+                    {
+                        return;
+                    }
+
+                    invalidMembersByValue.AddOrUpdate(fieldInitializerOperation.Value.ConstantValue.Value,
+                        new List<SyntaxNode> { fieldInitializerOperation.Syntax.Parent },
+                        (key, value) =>
+                        {
+                            value.Add(fieldInitializerOperation.Syntax.Parent);
+                            return value;
+                        });
+                }, OperationKind.FieldInitializer);
+
+                // ...and report at the end of the enum declaration
+                ssac.RegisterSymbolEndAction(sac =>
+                {
+                    foreach (var kvp in invalidMembersByValue)
+                    {
+                        foreach (var item in kvp.Value.OrderBy(x => x.GetLocation().SourceSpan).Skip(1))
```

Comment:
Is there anything better than location for the sorting to avoid reproducibility issues?

Reply:
I think location based sorting sounds the best approach.